liquibase-standard/src/main/java/liquibase/command/core/UpdateToTagCommandStep.java

```diff
@@ -87,35 +89,64 @@ public UpdateSummaryEnum getShowSummary(CommandScope commandScope) {
     }
 
     @Override
-    public ChangeLogIterator getStandardChangelogIterator(CommandScope commandScope, Database database, Contexts contexts, LabelExpression labelExpression, DatabaseChangeLog changeLog) throws DatabaseException {
+    public ChangeLogIterator getStandardChangelogIterator(CommandScope commandScope, Database database, Contexts contexts, LabelExpression labelExpression, DatabaseChangeLog changeLog) throws LiquibaseException {
         List<RanChangeSet> ranChangeSetList = database.getRanChangeSetList();
         String tag = commandScope.getArgumentValue(TAG_ARG);
 
         UpToTagChangeSetFilter upToTagChangeSetFilter = getUpToTagChangeSetFilter(tag, ranChangeSetList);
         if (! warningMessageShown && ! upToTagChangeSetFilter.isSeenTag()) {
-            checkForTagExists(changeLog, tag);
+            checkForTagExists(changeLog, tag, database);
         }
 
         List<ChangeSetFilter> changesetFilters = this.getStandardChangelogIteratorFilters(database, contexts, labelExpression);
         changesetFilters.add(upToTagChangeSetFilter);
         return new ChangeLogIterator(changeLog, changesetFilters.toArray(new ChangeSetFilter[0]));
     }
 
-    private void checkForTagExists(DatabaseChangeLog changeLog, String tag) {
+    private void checkForTagExists(DatabaseChangeLog changeLog, String tag, Database database) throws LiquibaseException {
         boolean found =
             changeLog.getChangeSets().stream().anyMatch(cs ->
                 cs.getChanges().stream().anyMatch(ch ->
                     ch instanceof TagDatabaseChange && ((TagDatabaseChange) ch).getTag().equals(tag)
                 )
             );
-        if (! found) {
+        if (!found && !GlobalConfiguration.STRICT.getCurrentValue()) {
             String message = String.format(
                     "The tag '%s' was not found in the changelog '%s'. All changesets in the changelog were deployed.%nLearn about options for undoing these changes at https://docs.liquibase.com.",
                     tag, changeLog.getPhysicalFilePath());
             Scope.getCurrentScope().getLog(UpdateToTagCommandStep.class).warning(message);
             Scope.getCurrentScope().getUI().sendMessage("WARNING:  " + message);
             warningMessageShown = true;
         }
+
+        if(GlobalConfiguration.STRICT.getCurrentValue()) {
+            TagDatabaseChange tagDatabaseChange = getTagDatabaseChange(changeLog.getChangeSets());
+            boolean thereIsTagDatabaseChange =  tagDatabaseChange != null;
```

Comment:
Yeah,  I saw it but decide to go with this change as it is, because `found` boolean flag considers both cases together (`tagDatabaseChange` exits and the runtime tag matches with the change's tag).